shapes.js

```diff
@@ -0,0 +1,348 @@
+function Shape(x, y, w, h, fill) {
```

Comment:
ES6 now supports default parameters, so no need for x||0. 
There are also now classes, so no need to use a function & its prototypes to simulate classes.